mne/io/base.py

```diff
@@ -831,13 +848,80 @@ def get_data(self, picks=None, start=0, stop=None,
         .. versionadded:: 0.14.0
         """
         picks = _picks_to_idx(self.info, picks, 'all', exclude=())
+
+        # Convert into the specified unit
+        ch_factors = np.ones(len(picks))
+
+        si_units = _handle_default('si_units')
+        si_units_splitted = {key: si_units[key].split('/') for key in si_units}
+        prefixes = {'': 1e0, 'd': 1e1, 'c': 1e2, 'm': 1e3, 'µ': 1e6, 'u': 1e6,
+                    'n': 1e9, 'p': 1e12, 'f': 1e15}
+        prefix_list = list(prefixes.keys())
+
+        if isinstance(units, str):
+            # Check that there is only one channel type
+            ch_types = self.get_channel_types(picks=picks, unique=True)
+            unit_ch_type = list(set(ch_types) & set(si_units.keys()))
+            if len(unit_ch_type) > 1:
+                raise ValueError('"units" cannot be str if there is more than '
+                                 'one channel type with a unit '
+                                 f'{unit_ch_type}.')
+            units = {unit_ch_type[0]: units}  # make the str argument a dict
+
+        if units is None:
+            pass
```

Comment:
I think I'd have expected something like this:

```python
if isinstance(units, None):
    pass
elif isinstance(units, str):
    ...
elif isinstance(units, dict):
    ...
else:
    raise ValueError()
```

Why are you doing:

```python
if isinstance(units, str):
    ...

if isinstance(units, None):
    pass
elif isinstance(units, dict):
    ...
else:
    raise ValueError()
```    

?

Also, you could save yourself one "else" and one "if" clause there by using `_validate_type`, see: https://github.com/mne-tools/mne-python/blob/bbcba957bcef31af9b49262ff1435a3c45c2e1df/mne/utils/check.py#L355-L365

Like this:

```python
_validate_type(units, types=(None, str, dict), item_name="units")
if str ...
elif dict ...

... no else necessary because we know it's NONE


```

Reply:
Thanks @sappelhoff 

> Why are you doing

This is the way I found to avoid having duplicate code in str and dict cases. What it's currently doing is if it's a string I create an appropriate dict from it. Then, if it was None I do nothing, elif it's a dict of units (either created or provided) I iterate over it to get conversion coefficients, else it's another type so not supported.

I guess I could also do instead:
```python
_validate_type(None, str, dict)
if str:
   convert to dict

# And after that:
if dict:  # originally provided or create from str
    loop over it and get coefficients
# if not it's necessarily None so nothing to do here so no need for an else
```

Reply:
I see, thanks. Yes, what you propose looks like less code to me that achieves the same thing while being just as readable.

But let's hear what @larsoner has to say before pushing any commits.

Reply:
The `if str:` then `if dict` code seems clean / readable and efficient to me